build fails with mingw32 version 4.9.1

Platform: Debian 8.6
Compiler: mingw32-gcc 4.9.1

```
src/tls/openssl/tls.c: In function ‘tls_set_selfsigned’:
src/tls/openssl/tls.c:251:13: error: ‘subj’ undeclared (first use in this function)
  X509_NAME *subj = NULL;
             ^
src/tls/openssl/tls.c:251:13: note: each undeclared identifier is reported only once for each function it appears in
src/tls/openssl/tls.c:252:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  EVP_PKEY *key = NULL;
  ^
make[1]: *** [build-i686/tls/openssl/tls.o] Error 1
make: *** [libre.a] Error 2
```

--- a/src/tls/openssl/tls.c
+++ b/src/tls/openssl/tls.c
@@ -24,6 +24,12 @@
 #include "tls.h"
 
 
+/* also defined by wincrypt.h */
+#ifdef WIN32
+#undef X509_NAME
+#endif
+
+
 #define DEBUG_MODULE "tls"
 #define DEBUG_LEVEL 5
 #include <re_dbg.h>

